src/parallel/parallel.jl

```diff
@@ -47,19 +47,14 @@ const MPI_IS_ROOT = Ref(true)
 
 @inline mpi_comm() = MPI.COMM_WORLD
 
-@inline mpi_rank(comm) = MPI.Comm_rank(comm)
 @inline mpi_rank() = MPI_RANK[]
 
-@inline n_mpi_ranks(comm) = MPI.Comm_size(comm)
 @inline n_mpi_ranks() = MPI_SIZE[]
 
-@inline is_parallel(comm) = n_mpi_ranks(comm) > 1
 @inline is_parallel() = MPI_IS_PARALLEL[]
```

Comment:
Some bikeshedding (that you also like :wink:): What about adding `mpi` to `is_parallel`, e.g. `is_mpi_parallel` (and `is_serial`)? This would make it even more clear that we're dealing with MPI here and not with, e.g. threaded parallelism. That choice would be consistent with `n_mpi_ranks` etc.

Reply:
I wouldn't consider this bikeshedding but rather striving for consistency, which I value even more, as you know ;-)

Thus, I like the idea. However, for the sake of consistency (and `grep`ability), we could also consider to *prefix* all these methods with `mpi_`, such that we would have:
```julia
@inline mpi_comm() = MPI.COMM_WORLD                                                                 
                                                                                                    
@inline mpi_rank() = MPI_RANK[]                                                                     
                                                                                                    
@inline mpi_n_ranks() = MPI_SIZE[]                                                                  
                                                                                                    
@inline mpi_is_parallel() = MPI_IS_PARALLEL[]                                                           
                                                                                                    
@inline mpi_is_serial() = MPI_IS_SERIAL[]                                                               
                                                                                                    
@inline mpi_is_root() = MPI_IS_ROOT[]                                                               
                                                                                                    
@inline mpi_root() = 0                                                                              
                                                                                                    
@inline mpi_println(args...) = is_mpi_root() && println(args...)                                    
@inline mpi_print(args...) = is_mpi_root() && print(args...)
```
Would that be nice (because consistency) or not as nice (due to not being properly "pronounceable" anymore)?

Reply:
Since we prefix many things already, prefixing these functions with `mpi_` sounds good to me. You could also think about following standards in `Base` and `Threads` by removing some additional `_`:
```julia
@inline mpi_comm() = MPI.COMM_WORLD                                                                 
                                                                                                    
@inline mpi_rank() = MPI_RANK[]                                                                     
                                                                                                    
@inline mpi_nranks() = MPI_SIZE[] # similar to Threads.nthreads()      
                                                                                                    
@inline mpi_isparallel() = MPI_IS_PARALLEL[] # similar to iszero, isapprox, isabspath, isreal, ...
                                                                                                    
@inline mpi_isserial() = MPI_IS_SERIAL[]                                                               
                                                                                                    
@inline mpi_isroot() = MPI_IS_ROOT[]                                                               
                                                                                                    
@inline mpi_root() = 0                                                                              
                                                                                                    
@inline mpi_println(args...) = is_mpi_root() && println(args...)                                    
@inline mpi_print(args...) = is_mpi_root() && print(args...)
```

Reply:
But it's also okay for me to keep the additional `_`.

Reply:
Great suggestion! Picked up and fixed in a7011af.